crossbeam-channel/src/flavors/never.rs

```diff
@@ -11,62 +11,62 @@ use crate::select::{Operation, SelectHandle, Token};
 use crate::utils;
 
 /// This flavor doesn't need a token.
-pub type NeverToken = ();
+pub(crate) type NeverToken = ();
 
 /// Channel that never delivers messages.
-pub struct Channel<T> {
+pub(crate) struct Channel<T> {
     _marker: PhantomData<T>,
 }
 
 impl<T> Channel<T> {
     /// Creates a channel that never delivers messages.
     #[inline]
-    pub fn new() -> Self {
+    pub(crate) fn new() -> Self {
```

Comment:
I'm hesitant about this change.  It seems `Channel` is already `pub(crate)`.  Then `pub(crate)` for its methods seems overkill.  What do you think?

Reply:
Indeed. If the struct is not public, then the method will never be public.

That said, there are cases where impl blocks are long, or the struct definitions are in different files, so I don't think it's really that bad to mark them as `pub(crate)` as unreachable_pub lint suggested.

Reply:
Okay... Let's follow rustc's suggestions. At worst, it's not part of the public API :)